TypeCobol.LanguageServer/Workspace.cs

```diff
@@ -573,20 +549,18 @@ public void DidChangeConfigurationParams(string[] arguments)
                 analyzerProviderWrapper.AddProvider(a);
             }
 
-            CompilationProject = new CompilationProject(_workspaceName, _rootDirectoryFullName, Helpers.DEFAULT_EXTENSIONS, Configuration.Format, typeCobolOptions, analyzerProviderWrapper);
+            // DidChangeConfiguration notification is always for the Default Workspace Project.
+            // The is for compatibility with for instance TypeCobol plugin and fallback.
+            this.WorkspaceProjectStore.DefaultWorkspaceProject.ReplaceCompilationProject(
+                Configuration, typeCobolOptions, analyzerProviderWrapper);
 
-            if (Configuration.CopyFolders != null && Configuration.CopyFolders.Count > 0)
-            {
-                foreach (var copyFolder in Configuration.CopyFolders)
-                {
-                    CompilationProject.SourceFileProvider.AddLocalDirectoryLibrary(copyFolder, false,
-                        new[] {".cpy"}, Configuration.Format.Encoding,
-                        Configuration.Format.EndOfLineDelimiter, Configuration.Format.FixedLineLength);
-                }                
-            }
+            // Update All WorkspaceProject instances because global configurations have change
+            this.WorkspaceProjectStore.UpdateProjectOptionsAndAnalyzerProvider(typeCobolOptions, analyzerProviderWrapper);
```

Comment:
For other projects than default, the format is not updated.

Let's consider what can change on a WorkspaceProject (and its underlying CompilationPrjecct) :
- on a configuration change
  - format
  - copy folders (for default project only)
  - analyzer provider
  - compilation options
- on a document configuration change
  - copy folders (for named projects)

To unify this, I think we should recreate CompilationProject instance for both default and named projects on configuration change and update the source file provider on document configuration change.